src/helm/benchmark/model_deployment_registry.py

```diff
@@ -0,0 +1,59 @@
+"""This allows users to register additional models via configuration file.
+
+TODO(#1673): Support adding other kinds of models besides AI21.
+"""
+
+from typing import Any, Dict, Iterable, Optional, List
+from dataclasses import dataclass
+
+import cattrs
+import yaml
+
+from helm.common.general import parse_hocon
+from helm.common.hierarchical_logger import hlog
+from helm.common.object_spec import ObjectSpec, create_object
+from helm.proxy.models import ALL_MODELS, FULL_FUNCTIONALITY_TEXT_MODEL_TAG, MODEL_NAME_TO_MODEL, TEXT_MODEL_TAG, Model
+
+
+class ClientSpec(ObjectSpec):
+    pass
+
+
+@dataclass(frozen=True)
+class ModelDeployment:
+    name: str
+    """Name of the model deployment."""
+
+    model_name: str
+    """Name of the model that this model deployment is for."""
+
+    client_spec: ClientSpec
+    """Specification for instantiating the client for this model deployment."""
+
+    max_sequence_length: Optional[int]
+    """Maximum equence length for this model deployment."""
+
+    tokenizer_name: Optional[str]
+    """Tokenizer for this model deployment."""
+
+
+@dataclass(frozen=True)
+class ModelDeployments:
+    model_deployments: List[ModelDeployment]
+
+
+_name_to_model_deployment: Dict[str, ModelDeployment] = {}
```

Comment:
Do we want to use a singleton here?  I believe schemas are passed around, which might make things more modular / easier to test?

Reply:
This may be the result of merging multiple configuration files (e.g. the repo's "defaults" configuration file + the user's or 3P repo's configuration files). So this has to be dynamically constructed. So we can't do the same thing as `schema.py`.